src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -96,8 +96,12 @@ public class WalletManager : IWalletManager
         // we keep a couple of objects in memory:
         // 1. the list of unspent outputs for checking whether inputs from a transaction are being spent by our wallet and
         // 2. the list of addresses contained in our wallet for checking whether a transaction is being paid to the wallet.
+        // 3. a mapping of all transactions in the wallet, to allow rapid lookup by txid
+        // 4. a mapping of all inputs with their corresponding transactions, to facilitate rapid lookup
         private Dictionary<OutPoint, TransactionData> outpointLookup;
         internal ScriptToAddressLookup scriptToAddressLookup;
+        private Dictionary<uint256, WalletIndexData> txLookup;
```

Comment:
This is not used, I'd say we should only add another index if we have a current use for it. also index by trx already exists in store via -txindex